### PR TITLE
Initial Game of Life Cheat sheet

### PR DESCRIPTION
**Pull request by thistleBgood**
Opening for feedback. Created at DuckDuckHack II Glasgow. Suggested by @kbremner

Future plans:
* Add animated GIFs demonstrating each of the four transition rules
* Eventually add interactive Game of Life (e.g. [this one](http://pmav.eu/stuff/javascript-game-of-life-v3.1.1/)).

https://duck.co/ia/view/game_of_life

### CODE REVIEW

**Comment by daxtheduck** [comment]
@thistleBgood Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/GameOfLife.pm [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package DDG::Goodie::GameOfLife;
```
Comment (MariagraziaAlastra) [inline lib/DDG/Goodie/GameOfLife.pm]: Hey @thistleBgood, you just need the JSON file for cheat sheets, so you can remove anything else :smile:

**Comment by MariagraziaAlastra** [comment]
Hey @thistleBgood, thanks for contributing, I just left a comment in the code :smile:

**Comment by zekiel** [comment]
Just for reference, @russellholt  is working on a full interactive IA for this: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/2167

File: share/goodie/cheat_sheets/json/game_of_life.json [hunk 2]
```diff
@@ -0,0 +1,146 @@
+{
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/game_of_life.json]: @thistleBgood can you please replace the `_` in the filename with `-`?

**Comment by moollaza** [comment]
@thistleBgood this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit another Instant Answer!

If you need ideas about how to contribute next, please feel free to reach out to me on Slack! I'm happy to chat about how you can help :+1:

**Comment by alohaas** [comment]
@thistleBgood, thanks again for creating the Game of Life Cheat Sheet! You’ve done a really good job here and we'd like you to help our community tackle the next big challenge – our mission to become the best search engine for programmers: https://duck.co/blog/post/297/help-for-programmers.

When you're ready to move on, please: 

- Create a new Instant Answer to increase our Programming IA coverage

OR

- Test live programming Instant Answers (or those in development) for bugs and suggest improvements 

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage).

Thanks again!

**Pull request merged**